src/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -131,19 +131,20 @@ A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
 """
 function NormalToricVariety( rays::Matrix{Int}, cones::Vector{Vector{Int}} )
     Incidence = Oscar.IncidenceMatrix(cones)
-    # arr = Polymake.@convert_to Array{Set{Int}} Polymake.common.rows(Incidence.pm_incidencematrix)
     pmntv = Polymake.fulton.NormalToricVariety(
         RAYS = Oscar.matrix_for_polymake(rays),
         MAXIMAL_CONES = Incidence,
     )
-    if length( cones ) == 1
-        return AffineNormalToricVariety( pmntv )
-    end    
-    return NormalToricVariety( pmntv )
+    return NormalToricVariety(pmntv)
 end
 
 export NormalToricVariety
 
+function AffineNormalToricVariety(v::NormalToricVariety)
+    isaffine(v) || error("Cannot construct affine toric variety from non-affine input")
+    return AffineNormalToricVariety(pm_ntv(v))
```

Comment:
If I understand the idea correctly, then the proposal is that methods specialized to affine normal toric varieties cannot be applied to `v::NormalToricVariety`, unless one explicitly casts `v` with the above method to a `AffineNormalToricVariety`.

This appears very clear, but also very rigid. We might want to reevaluate in the future, based on whether users like this rigid architecture or prefer a more flexible approach (e.g. casting automatically). In any case, for now I do not have any objections. Thank you @lkastner!